Dockerfile

```diff
@@ -8,15 +8,28 @@ RUN npm ci  --omit=dev --ignore-scripts
 
 COPY . .
 
-RUN npx prisma generate
-RUN npm run build
+RUN npx prisma generate && \
+    npm run build
 
 # Production image
 FROM node:${NODE_VERSION} AS production
 WORKDIR /usr/src/app
 
-COPY --from=builder /usr/src/app .
+COPY --from=builder /usr/src/app /usr/src/app
 
-EXPOSE 3000
+# Create a non-root user and group
+RUN groupadd -r appuser && useradd -r -g appuser -d /usr/src/app -s /sbin/nologin appuser
 
-CMD ["npm", "run", "start"]
+# Copy application files from builder
+COPY --from=builder /usr/src/app /usr/src/app
+
+# Change ownership of the application directory
+RUN chown -R appuser:appuser /usr/src/app
+
+# Switch to the non-root user
+USER appuser
+
+# Added healthcheck to satisfy checkov lint, you should configure this according to your application
+HEALTHCHECK --interval=30s --timeout=10s --start-period=5s --retries=3 CMD curl -f http://localhost:3000/health || exit 1
+
+CMD ["node", "dist/index.js"]
```

Comment:
All looks great 👍  I just have a question about this line, should it not stay at `npm run start`?

Reply:
I was verifying this docker file on another node.js script. Fixed it now.

Reply:
ok thank you 👍